azurelinuxagent/ga/exthandlers.py

```diff
@@ -478,10 +476,25 @@ def handle_ext_handlers(self, goal_state_id):
         max_dep_level = self.__get_dependency_level(all_extensions[-1]) if any(all_extensions) else 0
 
         depends_on_err_msg = None
+        extensions_enabled = conf.get_extensions_enabled()
         for extension, ext_handler in all_extensions:
 
             handler_i = ExtHandlerInstance(ext_handler, self.protocol, extension=extension)
 
+            # In case of extensions disabled, we skip processing extensions. But CRP is still waiting for some status
+            # back for the skipped extensions. In order to propagate the status back to CRP, we will report status back
+            # here with an error message.
+            if not extensions_enabled:
+                msg = "Extension will not be processed since extension processing is disabled. To enable extension " \
+                      "processing, set Extensions.Enabled=y in '/etc/waagent.conf'"
+                handler_i.set_handler_status(message=msg, code=-1)
```

Comment:
could you pass status=ExtHandlerStatusValue.not_ready explicitly? this parameter shouldn't have a default value in the first place, the status is very important and having a default makes the code harder to read (especially because 'code' defaults to 0, success, while status defaults to notready, failure)

we should remove those defaults, but on a different PR